apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+
+You have a few options:
+
+#### Attach named exports directly onto `exports`
+
+The "French 75" of packages: Classic but takes some sophistication and finesse.
+
+Pros:
+
+- Smaller package weight
+- Easy and simple (probably least effort if you don't mind keeping to a minor syntax stipulation)
+- Precludes the Dual-Package Hazard
+
+Cons:
+
+- Requires very specific syntax (either in source code and/or bundler gymnastics).
+
+**Working example**: [cjs-with-dual-distro (properties)](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/dual/property-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-source-with-esm-via-properties-distribution",
+  "main": "./dist/cjs/index.js"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-source-with-esm-via-properties-distribution",
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/cjs/index.js",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Typically, you would see `module.exports` assigned to something (be it an object or a function) like this:
+
+```cjs
+const someObject = {
+  foo() {},
+  bar() {},
+  qux() {},
+};
+
+module.exports = someObject;
+```
+
+Instead, do this:
+
+```cjs
+module.exports.foo = function foo() {};
+module.exports.foo = function bar() {};
+module.exports.foo = function qux() {};
+```
+
+#### Use a simple ESM wrapper
+
+The "Piña Colada" of packages: Complicated setup and difficult to get the balance right.
+
+Pros:
+
+- Smaller package weight
+
+Cons:
+
+- Likely requires complicated bundler gymnastics (we could not find any existing option to automate this in Webpack).
+
+**Working example**: [cjs-with-dual-distro (wrapper)](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/dual/wrapper-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-with-wrapper-dual-distro",
+  "exports": {
+    ".": {
+      "import": "./dist/esm/wrapper.mjs",
+      "require": "./dist/cjs/index.js",
+      "default": "./dist/cjs/index.js"
+    }
+  }
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-with-wrapper-dual-distro",
+  "type": "commonjs",
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": {
+      "import": "./dist/esm/wrapper.mjs",
+      "require": "./dist/cjs/index.js",
+      "default": "./dist/cjs/index.js"
+    },
+    "./package.json": "./package.json"
+  }
+}
+```
+
+In order to support named exports from the CJS bundle for an ESM consumer, this will need a bit of gymnastics from a bundler but is conceptually very simple.
+
+In certain conditions, CJS exports an object (which gets aliased to ESM's `default`); that object, like any object, is destructure-able. You can leverage that to pluck all the members of the object out, and then re-export them so the ESM consumer is none the wiser.
```

Comment:
Ooow! That is a good point. Destructuring does break the reference. Buuut I think this will not?

```js
import cjs from '../cjs/index.js';

export const a = cjs.a;
export const b = cjs.b;
export const c = cjs.c;
/* … */
```

Reply:
i believe it will still break it, because the "live binding" would be on the a, b, and c, which are `const`, and as such can't update anyways - that's not a live binding to `cjs.a`, for example.

Reply:
```js
const x = { a: 'hello' };

module.exports = x;

setTimeout(() => { x.a = 'bonjour' });
```
```js
export * from './index.cjs';
```
```js
import * as reexports from './reexport.mjs';

setTimeout(() => console.log(reexports));
/*
[Module: null prototype] {
  'module.exports': { a: 'bonjour' }
}
*/
```

That `'module.exports'` is quite unideal, but IIR, as designed—I can't remember the reason—it's very unergonomic, so I think it must have been a technical limitation?

Reply:
`module.exports` is new and is not something that users should rely on, it's a marker meant for bundlers, because there were some claims (https://github.com/nodejs/node/pull/53848) that adding this to the imported CJS facade helps bundlers that transpile CJS -> ESM, though I have yet to see any living bundlers actually proving that claim. As far as a non-bundler is concerned, this is as good as an empty object.

Reply:
Also, if the original CJS is assigning `exports.xxx`, then using an ESM wrapper that does `export *` won't help you with live binding either, because the CJS module facade uses a SyntheticModule that only evaluates and assigns the named exports once during evaluation, so it's as good as "export const a = cjs.a" - still broken.

Reply:
Ah, okay. So I think it is indeed impossible so best to document the limitation. There are better options anyway.